h2/src/main/org/h2/util/SourceCompiler.java

```diff
@@ -189,6 +226,24 @@ public Method getMethod(String className) throws ClassNotFoundException {
         return null;
     }
 
+    public Object invoke(String className, final Object... args) throws Exception {
+        String source = sources.get(className);
+        if (isJavaxScriptSource(source)) {
+            final Bindings bindings = new SimpleBindings();
+            int i = 0;
+            for (final Object arg : args) {
+                bindings.put("arg" + i, arg);
+                i++;
+            }
+            return this.getCompiledScript(className).eval(bindings);
+        } else {
+            final Method m = this.getMethod(className);
+            if (m.getParameterTypes().length != args.length)
+                throw new IllegalStateException("Wrong number of parameters, " + args.length + " were pased but " + m.getParameterTypes().length + " were needed");
```

Comment:
and check for args length != 0, since we don't allow args for triggers

Reply:
See above comment. I can however restore the more specific "No parameters are allowed for a trigger" exception if you want.

Reply:
Yeah, that's the part I want to keep. You can leave the return type as Object, per your comment, but please rename to getTrigger, since it is rather specific to the Trigger use-case.

Reply:
after your last change this invoke() method is now dead code

Reply:
yeah but I figured it might be useful to someone not versed in ScriptEngineManager. Do you want me to remove it ?

Reply:
Yes please. We can add it back if/when we need it. I don't like dead code lying around.